Engine/src/Engine/ResourceManagement/ShaderNorms.cpp

```diff
@@ -0,0 +1,105 @@
+#include <Engine/ResourceManagement/ShaderNorms.h>
+#include <Engine/ResourceManagement/ShaderGenerator.h>
+#include "Engine/ResourceManagement/ResourceManager.h"
+#include "glad/glad.h"
+
+using namespace std;
+
+namespace Engine
+{
+	const ShaderNorms::ShaderName ShaderNorms::DEFAULT_SHADER_NAME
+		= ShaderNorms::ShaderName::SN_TEXTURE_FILL;
+
+	ShaderNorms* ShaderNorms::m_singleton = nullptr;
+
+	ShaderNorms::ShaderNorms()
+	{
+		addShader(DEFAULT_SHADER_NAME);
+		addShader(ShaderName::SN_COLOR_FILL);
+		addShader(ShaderName::SN_GRADIENT_FILL);
+	}
+
+	ShaderNorms::~ShaderNorms()
+	{
+		for (auto [key, value] : m_shaders)
+		{
+			glDeleteProgram(value);
+		}
+	}
+
+	ShaderNorms* ShaderNorms::getInstance()
+	{
+		if (m_singleton == nullptr)
+		{
+			m_singleton = new ShaderNorms();
+		}
+		return m_singleton;
+	}
+
+	void ShaderNorms::update(int stride, int& textureCoordinates, int& colorCoordinates,
```

Comment:
This function is pretty important, right? It would be great to get some comments above the function itself saying what it does, and how/if you might call it as an engine developer

Reply:
Done!